commands/download.go

```diff
@@ -1,15 +1,119 @@
 package commands
 
 import (
+	"context"
+	"errors"
 	"fmt"
-	"github.com/jfrog/jfrog-cli-core/plugins/components"
-	"github.com/jfrog/jfrog-cli-core/utils/config"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
 	"github.com/jfrog/jfrog-client-go/utils/log"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
 )
 
-//nolint: unparam, lll
-func downloadSupportBundle(ctx *components.Context, rtDetails *config.ArtifactoryDetails, conf *supportBundleCommandConfiguration, response creationResponse) (string, error) {
-	tmpFile := "/tmp/foo.tmp"
-	log.Debug(fmt.Sprintf("Download Support Bundle %s from %s to %s", response.ID, rtDetails.Url, tmpFile))
-	return tmpFile, nil
+type downloadSupportBundleHttpClient interface {
+	GetURL() string
+	DownloadSupportBundle(bundleId string) (*http.Response, error)
+	GetSupportBundleStatus(bundleId string) (int, []byte, error)
+}
+
+func downloadSupportBundle(client downloadSupportBundleHttpClient, timeout time.Duration, retryInterval time.Duration,
+	bundleId string) (string, error) {
+
+	dirPath, err := fileutils.CreateTempDir()
```

Comment:
Create temp file once the status check is OK or when download is OK.

Reply:
fix